src/Project/ToolRunner.ts

```diff
@@ -25,29 +25,59 @@ const which = require('which');
 const K_DATA: string = 'data';
 const K_EXIT: string = 'exit';
 
+// successfully killed by calling kill() method
+export type SelfKilled = 'SelfKilled';
 
 export class ToolRunner {
   tag = this.constructor.name;  // logging tag
 
+  private child: cp.ChildProcessWithoutNullStreams|undefined = undefined;
+
+  // When the spawned process was killed by kill() method, set this true
+  private killedByMe = false;
+
   private handlePromise(
-      resolve: (value: string|PromiseLike<string>) => void,
-      reject: (value: string|PromiseLike<string>) => void, cmd: cp.ChildProcessWithoutNullStreams) {
+      resolve: (value: string|SelfKilled|PromiseLike<string>) => void,
+      reject: (value: string|NodeJS.Signals|PromiseLike<string>) => void) {
     // stdout
-    cmd.stdout.on(K_DATA, (data: any) => {
+    this.child!.stdout.on(K_DATA, (data: any) => {
       Logger.append(data.toString());
     });
     // stderr
-    cmd.stderr.on(K_DATA, (data: any) => {
+    this.child!.stderr.on(K_DATA, (data: any) => {
       Logger.append(data.toString());
     });
 
-    cmd.on(K_EXIT, (code: any) => {
+    this.child!.on(K_EXIT, (code: number|null, signal: NodeJS.Signals|null) => {
+      this.child = undefined;
+
+      // From https://nodejs.org/api/child_process.html#event-exit
+      //
+      // The 'exit' event is emitted after the child process ends.
+      // If the process exited, code is the final exit code of the process, otherwise null.
+      // If the process terminated due to receipt of a signal, signal is the string name
+      // of the signal, otherwise null.
+      // One of the two will always be non-null.
+
+      // when child was terminated due to a signal
+      if (code === null) {
+        Logger.debug(this.tag, `Child process was killed (signal: ${signal!})`);
+
+        if (this.killedByMe) {
+          resolve('SelfKilled');
+        } else {
+          reject(signal!);
+        }
+        return;
```

Comment:
This `return` is unnecessary.

Reply:
This is necessary. L63 to L71 handles when `code === null`. Code from L75 handles when `code !== null`

Reply:
Sorry for the very late reply. Now I read this comment. I'd like to apologize. 😭 

My meaning is like that

as-is
```ts
      if (code === null) {
        Logger.debug(this.tag, `Child process was killed (signal: ${signal!})`);

        if (this.killedByMe) {
          resolve('SelfKilled');
        } else {
          reject(signal!);
        }
        return;
      }
```

to-be
```ts
      if (code === null) {
        Logger.debug(this.tag, `Child process was killed (signal: ${signal!})`);

        if (this.killedByMe) {
          return resolve('SelfKilled');
        } else {
          return reject(signal!);
        }
      }
```